src/makeRelativeDate.ts

```diff
@@ -0,0 +1,164 @@
+type RelativeDateOptions = {
+	notAfter?: number;
+	format?: FormatType;
+	showTime?: boolean;
+};
+
+type FormatType = 'short' | 'med' | 'long';
+type PeriodType = 's' | 'm' | 'h' | 'd';
+
+const dayOfWeek = (day: number): string =>
+	[
+		'Sunday',
+		'Monday',
+		'Tuesday',
+		'Wednesday',
+		'Thursday',
+		'Friday',
+		'Saturday',
+	][day];
+
+const monthAbbr = (month: number): string =>
+	[
+		'Jan',
+		'Feb',
+		'Mar',
+		'Apr',
+		'May',
+		'Jun',
+		'Jul',
+		'Aug',
+		'Sep',
+		'Oct',
+		'Nov',
+		'Dec',
+	][month];
+
+const pad = (n: number): number | string => (n < 10 ? `0${n}` : n);
+
+const isToday = (date: Date): boolean => {
+	const today = new Date();
+	return date.toDateString() === today.toDateString();
+};
+
+const isWithin24Hours = (date: Date): boolean => {
+	const today = new Date();
+	return date.valueOf() > today.valueOf() - 24 * 60 * 60 * 1000;
+};
+
+const isYesterday = (relative: Date): boolean => {
+	const today = new Date();
+	const yesterday = new Date();
+	yesterday.setDate(today.getDate() - 1);
+	return relative.toDateString() === yesterday.toDateString();
+};
+
+const isWithinPastWeek = (date: Date): boolean => {
+	const weekAgo = new Date().valueOf() - 7 * 24 * 60 * 60 * 1000;
+	return date.valueOf() >= weekAgo;
+};
+
+const isValidDate = (date: Date): boolean => {
```

Comment:
we have some other `is` methods exported in libs, how do you feel about pulling this out into its own module?

Reply:
Nice idea. I'll do that in a follow up PR. Carded up here: https://trello.com/c/EvexVKCw/2734-move-isvaliddate-into-its-own-helper-util-in-libs

Reply:
nice one